packages/math-input/src/components/__tests__/integration.test.tsx

```diff
@@ -0,0 +1,183 @@
+import "@testing-library/jest-dom";
+import {screen, render, fireEvent, within} from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import React, {useState} from "react";
+
+import MathInput from "../input/math-input";
+import KeypadContext from "../keypad-context";
+import KeypadSwitch from "../keypad-switch";
+
+import type {KeypadAPI} from "../../types";
+
+function InputWithContext() {
+    const [value, setValue] = useState<string>("");
+
+    return (
+        <KeypadContext.Consumer>
+            {({keypadElement}) => {
+                return (
+                    <MathInput
+                        keypadElement={keypadElement as any}
+                        value={value}
+                        onChange={(nextValue, cb) => {
+                            setValue(nextValue);
+                            cb();
+                        }}
+                        onFocus={() => {
+                            keypadElement?.activate();
+                        }}
+                        onBlur={() => {
+                            keypadElement?.dismiss();
+                        }}
+                    />
+                );
+            }}
+        </KeypadContext.Consumer>
+    );
+}
+
+function KeypadWithContext() {
+    return (
+        <KeypadContext.Consumer>
+            {({setKeypadElement}) => {
+                return (
+                    <KeypadSwitch
+                        onElementMounted={setKeypadElement}
+                        onDismiss={() => {}}
+                        onAnalyticsEvent={async () => {}}
+                        useV2Keypad
+                    />
+                );
+            }}
+        </KeypadContext.Consumer>
+    );
+}
+
+function ConnectedMathInput() {
+    const [keypadElement, setKeypadElement] = useState<KeypadAPI | null>();
+    const [renderer, setRenderer] = useState<any>(null);
+    const [scrollableElement, setScrollableElement] =
+        useState<HTMLElement | null>();
+
+    return (
+        <KeypadContext.Provider
+            value={{
+                setKeypadElement,
+                keypadElement,
+                setRenderer,
+                renderer,
+                setScrollableElement,
+                scrollableElement,
+            }}
+        >
+            <InputWithContext />
+            <KeypadWithContext />
+        </KeypadContext.Provider>
+    );
+}
+
+describe("math input integration", () => {
+    it("renders", () => {
+        render(<ConnectedMathInput />);
+
+        expect(
+            screen.getByLabelText(
+                "Math input box Tap with one or two fingers to open keyboard",
```

Comment:
I wonder if we should fix this label text as it seems messy. I wonder how it reads out in a screen reader. I know you didn't change this as part of your PR, just noticing that this looks strange. :)

Reply:
Mostly it seems like it needs some punctuation. "Math input box. Tap with one or two fingers to open keyboard."

Reply:
I think that's a good point; nervous about doing that here though since I think that would then need to go through the translation pipeline. 🤔